### ISSUE TITLE
Setup with latest elixir is broken

### ISSUE THREAD

**Issue by dholdren** [stride-nyc/remote_retro#477]
As a developer,
I want to be able to setup and install remote_retro locally using latest elixir/mix/hex
So that I can work on a fork or install on a server.

__Acceptance Criteria:__
-  Following the setup instructions in the README.md as of 8/15/2018 should just work
and not produce the following error:
```
➜  remote_retro git:(master) ✗ mix

Greetings Developer,
  Looks like your local Hex is outdated. Please update your local Hex via `mix local.hex` and try again.

  Thanks!

[1]    70326 abort      mix
➜  remote_retro git:(master) ✗ mix local.hex

Found existing entry: /Users/dean.holdren/.mix/archives/hex-0.18.1
Are you sure you want to replace it with "https://repo.hex.pm/installs/1.6.0/hex-0.18.1.ez"? [Yn] Y
* creating /Users/dean.holdren/.mix/archives/hex-0.18.1
➜  remote_retro git:(master) ✗ mix

  Greetings Developer,
  Looks like your local Hex is outdated. Please update your local Hex via `mix local.hex` and try again.

  Thanks!

[1]    55217 abort      mix
```

I'm new to Elixir, but it seems like we need a way to lock versions of certain things. I found this code in https://github.com/stride-nyc/remote_retro/blob/master/lib/tasks/preflight.ex#L27-L41
```
defp check_hex_version do
    required_version_num = ~r/0\.17\.\d+/
```

**Comment by vanderhoop** [comment]
Hey @dholdren, sorry you fell victim to this. It's related to #403. 

Right now, we're best setup for Elixir 1.5, and I think checking for that would make a smarter preflight check than the Hex version. Alternatively, if you can get the application to blend on elixir your current elixir version (looks like 1.6.0) with Hex 0.18.1, by all means bump the version requirements in mix.exs and prelight.exs and open a PR!

**Comment by dplain90** [comment]
@vanderhoop @dholdren hit this issue too, took me a while to backtrack and get the erlang/elixir versions to match up after I had installed newer versions.

However, no matter what I did - even with the correct setup of Elixir 1.5 and Erl 20, it was still using hex-0.18. I had to change the regex in check_hex_version to get it to work. Anyways, I can confirm that the setup of: 

- v1.5 Elixir
- v20 Erlang
with hex-0.18 

I think you're right that it'd be better to match Elixir. Also probably a check for Erlang version too because that tripped me up as well.
